hail_search/queries/mito.py

```diff
@@ -112,6 +118,215 @@ class MitoHailTableQuery(BaseHailTableQuery):
+
+            wes_ht = wes_ht.rename({'family_entries': WES_FAMILY_ENTRIES_FIELD, 'filters': WES_FILTERS_FIELD})
+            wgs_ht = wgs_ht.rename({'family_entries': WGS_FAMILY_ENTRIES_FIELD, 'filters': WGS_FILTERS_FIELD})
+            ht = wes_ht.join(wgs_ht, how='outer')
+
+            # Filter by quality
+            ht = self._annotate_passes_quality_filter(ht, quality_filter, **kwargs)
+
+            # Filter by inheritance
+            any_valid_entry, is_any_affected = self._get_any_family_member_gt_has_alt_filter()
+            ht = ht.annotate(wes_passes_inheritance=ht[WES_FAMILY_ENTRIES_FIELD].map(
+                lambda entries: hl.or_missing(entries.any(any_valid_entry), entries)
+            ))
+            ht = ht.annotate(wgs_passes_inheritance=ht[WGS_FAMILY_ENTRIES_FIELD].map(
+                lambda entries: hl.or_missing(entries.any(any_valid_entry), entries)
+            ))
+
+            if self._has_comp_het_search:
+                comp_het_ht = self._annotate_families_inheritance(ht, COMPOUND_HET, inheritance_filter, sorted_wes_family_sample_data, SAMPLE_TYPE_WES)
+                comp_het_ht = self._annotate_families_inheritance(comp_het_ht, COMPOUND_HET, inheritance_filter, sorted_wgs_family_sample_data, SAMPLE_TYPE_WGS)
+                comp_het_ht = self._apply_filters(comp_het_ht, family_idx_map)
+                filtered_comp_het_hts.append(comp_het_ht)
+
+            if is_any_affected or not (inheritance_filter or self._inheritance_mode):
+                # No sample-specific inheritance filtering needed
+                sorted_wes_family_sample_data = sorted_wgs_family_sample_data = []
+
+            if self._inheritance_mode != COMPOUND_HET:
+                ht = self._annotate_families_inheritance(ht, self._inheritance_mode, inheritance_filter, sorted_wes_family_sample_data, SAMPLE_TYPE_WES)
+                ht = self._annotate_families_inheritance(ht, self._inheritance_mode, inheritance_filter, sorted_wgs_family_sample_data, SAMPLE_TYPE_WGS)
+                ht = self._apply_filters(ht, family_idx_map)
+                filtered_hts.append(ht)
+
+        merged_ht = self._merge_project_hts(filtered_hts, n_partitions)
+        merged_comp_het_ht = self._merge_project_hts(filtered_comp_het_hts, n_partitions)
+        return merged_ht, merged_comp_het_ht
+
+    @staticmethod
+    def _build_family_index_map(sorted_wes_family_sample_data, sorted_wgs_family_sample_data):
+        family_guid_idx_map = defaultdict(dict)
+
+        for family_idx, samples in enumerate(sorted_wes_family_sample_data):
+            family_guid = samples[0]['familyGuid']
+            family_guid_idx_map[family_guid][SAMPLE_TYPE_WES] = family_idx
+
+        for family_idx, samples in enumerate(sorted_wgs_family_sample_data):
+            family_guid = samples[0]['familyGuid']
+            family_guid_idx_map[family_guid][SAMPLE_TYPE_WGS] = family_idx
+
+        return hl.literal(family_guid_idx_map)
```

Comment:
Is there a reason to use a generic `literal` instead of making it a dictionary?

Reply:
good catch, using hl.dict instead here now.